File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.go
import (
    "math"
)

func increasingTriplet(nums []int) bool {
    low, mid := math.MaxInt32, math.MaxInt32
    for _, n := range nums {
        if n <= low {
            low = n
        } else if n <= mid {
            mid = n
        } else {
            return true
        }
    }
    return false
}

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.c
bool increasingTriplet(int* nums, int numsSize){
    int i, low, mid;
    low = mid = INT_MAX;
    
    for (i = 0; i < numsSize; i++) {
        if (nums[i] <= low)
            low = nums[i];
        else if (nums[i] <= mid) 
            mid = nums[i];
        else
            return true;
    }
    return false;
}